src/components/10-atoms/checkbox/index.scss

```diff
@@ -3,130 +3,171 @@ $checkbox-border-width: 1px;
-      border: $checkbox-border-width solid;
-      border-color: $color-prim-gray-dusty;
-      box-sizing: border-box;
+    &:disabled + .a-checkbox__icon--inverted {
+      background-color: transparent;
+      border-color: $color-disabled-inverted;
 
-      &:not(.a-checkbox__icon--checkmark) {
-        &::after {
-          content: '';
-          display: block;
-          position: absolute;
-          top: ($checkbox-dimension - $checkbox-icon-dimension - 2 * $checkbox-border-width) / 2;
-          left: ($checkbox-dimension - $checkbox-icon-dimension - 2 * $checkbox-border-width) / 2;
-          width: $checkbox-icon-dimension;
-          height: $checkbox-icon-dimension;
-          border: 1px solid;
-          border-color: transparent;
-          box-sizing: border-box;
-        }
+      .a-checkbox__icon-checkmark {
+        color: $color-disabled-inverted;
       }
     }
+  }
 
-    .a-checkbox__icon-checkmark {
+  .a-checkbox__icon {
+    display: block;
+    position: relative;
+    transition: all 0.2s ease;
+    width: $checkbox-dimension;
+    height: $checkbox-dimension;
+    border: $checkbox-border-width solid;
+    border-color: $color-prim-gray-dusty;
+    box-sizing: border-box;
+
+    &-checkmark {
       display: none;
     }
 
-    .a-checkbox__content {
-      display: block;
-      margin: -19px 0 0 30px;
-      line-height: 1.37;
-      color: $color-prim-gray-mine-shaft;
+    &--inverted {
+      border-color: $color-prim-white;
     }
 
-    .a-checkbox__error {
-      display: block;
-      margin: $checkbox-error-top-margin 0 0 ($checkbox-dimension + $checkbox-label-left-margin);
-      line-height: 1.37;
+    &:not(.a-checkbox__icon--checkmark) {
+      &::after {
+        content: '';
+        display: block;
+        position: absolute;
+        top: ($checkbox-dimension - $checkbox-icon-dimension - 2 * $checkbox-border-width) / 2;
+        left: ($checkbox-dimension - $checkbox-icon-dimension - 2 * $checkbox-border-width) / 2;
+        width: $checkbox-icon-dimension;
+        height: $checkbox-icon-dimension;
+        border: 1px solid;
+        border-color: transparent;
+        box-sizing: border-box;
+      }
     }
+  }
 
-    &[disabled] {
-      .a-checkbox__content {
-        color: $color-prim-gray-dusty;
-      }
+  .a-checkbox__content {
+    display: block;
+    margin: -19px 0 0 30px;
+    line-height: 1.37;
+    color: $color-prim-gray-mine-shaft;
 
-      .a-checkbox__input:checked + .a-checkbox__icon::after {
-        border-color: $color-prim-gray-silver;
-        background-color: $color-prim-gray-silver;
-      }
+    &--inverted {
+      color: $color-prim-white;
     }
+  }
 
-    &[error] {
-      color: $color-validation-error;
-  
-      &:not([checked]) .a-checkbox__icon {
-          border-color: $color-validation-error;
+  .a-checkbox__error {
+    display: block;
+    margin: $checkbox-error-top-margin 0 0 ($checkbox-dimension + $checkbox-label-left-margin);
```

Comment:
I would not use the combined margin property here, since this thing becomes unreadable.

Reply:
thats not my code change